persistence-modules/spring-persistence-simple-2/src/test/java/com/baeldung/jndi/datasource/SimpleJNDIUnitTest.java

```diff
@@ -0,0 +1,37 @@
+package com.baeldung.jndi.datasource;
+
+import static org.junit.Assert.assertEquals;
+
+import javax.naming.Context;
+import javax.naming.InitialContext;
+import javax.sql.DataSource;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class SimpleJNDIUnitTest {
+
+    private InitialContext initContext;
+
+    @BeforeEach
+    public void setup() throws Exception {
+        System.setProperty("java.naming.factory.initial", "org.osjava.sj.SimpleContextFactory");
```

Comment:
If we are going to use System properties shouldn't we reset them when the test tears down? We don't want any unexpected side effects when other tests run.

Reply:
Done!